lib/redis/client.rb

```diff
@@ -415,7 +415,7 @@ def _parse_options(options)
       end
 
       url = options[:url]
-      url = defaults[:url] if url.nil?
+      url = defaults[:url] if url.nil? && options.empty?
```

Comment:
I'm afraid this might break it for many users. Especially ones relying on the default URL, but adding options like `timeout` etc.

Reply:
Good point.

To this is how it could work:
- if the options overlap with any information that can be derived from REDIS_URL, use options as the source of truth and ignore REDIS_URL
- when all options and information derived from REDIS_URL can coexist, put them together

Partially overwriting information from the REDIS_URL is a rather strange behavior IMHO

Reply:
> Partially overwriting information on top of the REDIS_URL is a rather strange behavior IMHO

Yeah, there is already a bunch of open issues about that. Nobody is really happy with it, but it's almost impossible to change it without breaking meany users.

So it will have to yet for the next major.